python/cudf/cudf/core/dataframe.py

```diff
@@ -820,11 +825,39 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
+            other_cols = other.to_pandas().to_dict()
+            other_cols_keys = list(other_cols.keys())
+            result_cols = list(self.columns)
+            df_cols = list(result_cols)
+            for new_col in other_cols.keys():
+                if new_col not in result_cols:
+                    result_cols.append(new_col)
+            for col in result_cols:
+                if col in df_cols and col in other_cols_keys:
+                    l_opr = self._cols[col]
+                    r_opr = other_cols[col]
+                else:
+                    if col not in df_cols:
+                        r_opr = other_cols[col]
+                        l_opr = Series(
+                            column_empty(
+                                len(self),
+                                masked=True,
+                                dtype=np.array(r_opr).dtype,
+                            )
+                        )
+                    if col not in other_cols_keys:
+                        r_opr = None
+                        l_opr = self._cols[col]
+                result[col] = op(l_opr, r_opr)
+            # hack to replicate pandas alignment order
+            permute_cols = (
+                pd.Series(index=df_cols)
+                .align(pd.Series(index=other_cols_keys))[0]
+                .index
             )
```

Comment:
I'm not following the issue here, could you explain it?

Reply:
Pandas sorts the columns in the result like this: Numeric column names first, followed by bools, then strings at the end, regardless of the way the columns are ordered in the operands. A result dataframe with a mix of column names/types would be sorted like this:

`[0,1,2,'1','12','20', 'a','b','c']`

I found this behavior pretty hard to replicate with a simple lambda or anything, and ended up with the pandas function.

Reply:
Sounds like this is unintended behavior that we shouldn't necessarily replicate. Any opposition?

Reply:
No opposition, I would rather ignore it too. I'll just xfail that.

Reply:
In addition to an xfail, I'd add a test to make sure we still get the results correct even if the columns are in a different order by just realigning the columns in the test based on either cudf or pandas output.

Reply:
Added.